app/master/subjob.py

```diff
@@ -40,11 +41,15 @@ def _set_atom_state(self, state):
         for atom in self._atoms:
             atom.state = state
 
-    def mark_in_progress(self):
+    def mark_in_progress(self, slave):
```

Comment:
It's about the same I'd say. You never know in the future when we might need a back reference to the Slave for other stuff.